java/base-client/src/main/java/no/bankaxept/epayment/client/base/accesstoken/ScheduledAccessTokenPublisher.java

```diff
@@ -108,12 +105,82 @@ public void shutDown() {
     @Override
     public void subscribe(Flow.Subscriber<? super String> subscriber) {
         var token = atomicToken.get();
-        if (token != null && token.getExpiry().isBefore(clock.instant()))
+        if (token != null){
+            if(token.getExpiry().isBefore(clock.instant())) {
+                var exception = new IllegalStateException("Token already expired at: " + token.getExpiry());
+                subscriber.onError(exception);
+                onError(exception);
+
+            }
             subscriber.onNext(token.getToken());
+        }
         else {
             synchronized (subscribers) {
                 subscribers.add(subscriber);
             }
         }
     }
+
+    public static class Builder {
+        private String uri;
+        private HttpClient httpClient;
+        private ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+        private Clock clock = Clock.systemDefaultZone();
+
+        private final LinkedHashMap<String, List<String>> headers = new LinkedHashMap<>();
+        private final StringBuilder body = new StringBuilder();
+
+        public Builder httpClient(HttpClient httpClient) {
+            this.httpClient = httpClient;
+            return this;
+        }
+
+        public Builder uri(String uri) {
+            this.uri = uri;
+            return this;
+        }
+
+        public Builder apimKey(String apimKey) {
+            headers.put("Ocp-Apim-Subscription-Key", List.of(apimKey));
+            return this;
+        }
+
+        public Builder credentials(String id, String secret) {
+            headers.put("Authorization", List.of("Basic " + Base64.getEncoder().encodeToString((id + ":" + secret).getBytes(StandardCharsets.UTF_8))));
+            return this;
+        }
+
+        public Builder grantType(GrantType grantType) {
+            prefix().append("grant_type=").append(grantType);
+            return this;
+        }
+
+        public Builder scopes(List<Scope> scopes) {
```

Comment:
`OAuth does not define any particular values for scopes, since it is highly dependent on the service's internal architecture and needs.`.
That's why I did it like this, and implemented BankIDScope in platform project

Reply:
Why not just use a String in this generic code, then? The interface is just a supplier now. We can still use enums on the consumers.

Reply:
Yeah, makes sense. Changed it to string